includes/sanitizers/class-amp-style-sanitizer.php

```diff
@@ -2661,23 +2691,42 @@ private function process_font_face_at_rule( AtRuleSet $ruleset, $options ) {
 					'genericons.woff',
 				];
 				if ( in_array( $font_filename, $bundled_fonts, true ) ) {
-					$data_url->getURL()->setString( plugin_dir_url( AMP__FILE__ ) . "assets/fonts/$font_filename" );
-					$converted_count++;
+					$font_file = plugin_dir_url( AMP__FILE__ ) . "assets/fonts/$font_filename";
+					$data_url->getURL()->setString( $font_file );
+					$font_files[] = $font_file;
 				}
 			} // End foreach $source_data_url_objects.
 		} // End foreach $src_properties.
 
-		/*
-		 * If a data: URL has been replaced with an external file URL, then we add a font-display:swap to the @font-face
-		 * rule if one isn't already present. This prevents FO
-		 *
-		 *  If no font-display is already present, add font-display:swap since the font is now being loaded externally.
+		/**
+		 * Override the 'font-display' property to improve font performance.
 		 */
-		if ( $converted_count && 0 === count( $ruleset->getRules( 'font-display' ) ) ) {
+		if ( isset( $font_family ) && in_array( $font_family, array_keys( $this->args['font_face_display_overrides'] ), true ) ) {
+			$properties = $ruleset->getRules( 'font-display' );
+			if ( isset( $properties[0] ) ) {
+				$ruleset->removeRule( $properties[0] );
+			}
+
 			$font_display_rule = new Rule( 'font-display' );
-			$font_display_rule->setValue( 'swap' );
+			$font_display_rule->setValue( $this->args['font_face_display_overrides'][ $font_family ] );
 			$ruleset->addRule( $font_display_rule );
 		}
+
+		/**
+		 * If the font-display is auto, block, or swap then we should automatically add the preload link for the first font file.
+		 */
+		$properties = $ruleset->getRules( 'font-display' );
+		if (
+			(
+				( isset( $properties[0] ) && in_array( $properties[0]->getValue(), [ 'auto', 'block', 'swap' ], true ) )
+				||
+				( ! isset( $properties[0] ) ) // Defaults to 'auto', hence should be preloaded as well.
+			)
+			&&
+			1 <= count( $font_files )
+		) {
+			$this->dom->links->addPreload( $font_files[0], RequestDestination::FONT );
+		}
```

Comment:
I just read [Prevent layout shifting and flashes of invisible text (FOIT) by preloading optional fonts](https://web.dev/preload-optional-fonts/) which recommends adding preload links to optional fonts as well, although it says:

> Although it is not necessary to preload an optional font, it greatly improves the chance for it to load before the first render cycle, especially if it is not yet stored in the browser's cache.

Nevertheless, adding a preload will introduce network congestion as resources compete for bandwidth. So if the font truly is optional, IMO it seems preferable to not preload it.